sql-plugin/src/main/311until330-nondb/scala/org/apache/spark/sql/execution/datasources/parquet/ShimCurrentBatchIterator.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.parquet
+
+import java.io.IOException
+import java.lang.reflect.Method
+
+import com.nvidia.spark.CurrentBatchIterator
+import com.nvidia.spark.rapids.ParquetCachedBatch
+import java.util
+import org.apache.hadoop.conf.Configuration
+import org.apache.parquet.ParquetReadOptions
+import org.apache.parquet.column.ColumnDescriptor
+import org.apache.parquet.schema.Type
+
+import org.apache.spark.TaskContext
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.execution.datasources.parquet.rapids.shims.ShimVectorizedColumnReader
+import org.apache.spark.sql.execution.vectorized.WritableColumnVector
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.internal.SQLConf.LegacyBehaviorPolicy
+
+object ParquetVectorizedReader {
+  private var readBatchMethod: Method = null
+  def getReadBatchMethod(): Method = {
+    if (readBatchMethod == null) {
+      readBatchMethod =
+        classOf[VectorizedColumnReader].getDeclaredMethod("readBatch", Integer.TYPE,
+          classOf[WritableColumnVector])
+      readBatchMethod.setAccessible(true)
```

Comment:
What is the advantage of using `MethodUtils`? 
It doesn't reduce the numbers of line of code
```
val method = MethodUtils.getMatchingMethod(classOf[VectorizedColumnReader], "readBatch", Integer.TYPE,  classOf[WritableColumnVector])
method.setAccessible(true)
```
Nor do we run in JRE prior to 1.4 where `MethodUtils` has a work around.

IMO this would bring a 3rd party in to the picture when it's not really adding any value. Please feel free to elaborate on why you made the suggestion.

Reply:
not a big deal but there are invokeMethod flavors where you can just pass forceAccess = true

Reply:
That will look up the method every time as seen [here](https://github.com/apache/commons-lang/blob/80889fc45cbfb3da0a916e0fc0d51663bb970c2a/src/main/java/org/apache/commons/lang3/reflect/MethodUtils.java#L205). I don't think that's what we want